core/src/main/java/org/apache/iceberg/encryption/Ciphers.java

```diff
@@ -19,19 +19,29 @@
 
 package org.apache.iceberg.encryption;
 
+import java.nio.charset.StandardCharsets;
 import java.security.GeneralSecurityException;
 import java.security.SecureRandom;
 import javax.crypto.AEADBadTagException;
 import javax.crypto.Cipher;
 import javax.crypto.spec.GCMParameterSpec;
 import javax.crypto.spec.SecretKeySpec;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.primitives.Ints;
 
 public class Ciphers {
-  private static final int NONCE_LENGTH = 12;
-  private static final int GCM_TAG_LENGTH = 16;
+  public static final int NONCE_LENGTH = 12;
+  public static final int GCM_TAG_LENGTH = 16;
+  public static final String GCM_STREAM_MAGIC_STRING = "GCM1";
```

Comment:
Where does this come from? Is it introduced by Iceberg or from another standard?

Reply:
The GCM Streams are introduced in this PR - being proposed as a part of the Iceberg spec / standard.